sql/colcontainer: TestDiskQueue/Rewindable/DiskQueueCacheMode=0/AlwaysCompress=true/NumBatches=999 timed out

[(sql/colcontainer).TestDiskQueue/Rewindable/DiskQueueCacheMode=0/AlwaysCompress=true/NumBatches=999 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2162409&tab=buildLog) on [master@f44f0c40dd7680d3425185acc2397d8a8165fe38](https://github.com/cockroachdb/cockroach/commits/f44f0c40dd7680d3425185acc2397d8a8165fe38):

```
Slow failing tests:
TestDiskQueue/Rewindable/DiskQueueCacheMode=0/AlwaysCompress=true/NumBatches=999 - 0.00s

Slow passing tests:
<none>

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestDiskQueue/Rewindable/DiskQueueCacheMode=0/AlwaysCompress=true/NumBatches=999 PKG=./pkg/sql/colcontainer TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestDiskQueue%2FRewindable%2FDiskQueueCacheMode%3D0%2FAlwaysCompress%3Dtrue%2FNumBatches%3D999.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

```
=== RUN   TestDiskQueue
=== RUN   TestDiskQueue/DiskQueueCacheMode=0/AlwaysCompress=true/BufferSizeBytes=0_B/MaxFileSizeBytes=10_KiB/NumBatches=720
=== RUN   TestDiskQueue/DiskQueueCacheMode=0/AlwaysCompress=false/BufferSizeBytes=0_B/MaxFileSizeBytes=7.2_MiB/NumBatches=26
=== RUN   TestDiskQueue/DiskQueueCacheMode=1/AlwaysCompress=true/BufferSizeBytes=173_KiB/MaxFileSizeBytes=10_KiB/NumBatches=744
=== RUN   TestDiskQueue/DiskQueueCacheMode=1/AlwaysCompress=false/BufferSizeBytes=173_KiB/MaxFileSizeBytes=11_MiB/NumBatches=995
=== RUN   TestDiskQueue/Rewindable/DiskQueueCacheMode=1/AlwaysCompress=true/NumBatches=835
=== RUN   TestDiskQueue/Rewindable/DiskQueueCacheMode=0/AlwaysCompress=true/NumBatches=999
panic: test timed out after 40m0s
```
```
goroutine 16 [runnable]:
fmt.(*fmt).fmtSbx(0xc11f3a3f20, 0x0, 0x0, 0xc127fb1500, 0x63f, 0x700, 0x2d71496, 0x11)
	/usr/local/go/src/fmt/format.go:425 +0x53d
fmt.(*fmt).fmtBx(...)
	/usr/local/go/src/fmt/format.go:441
fmt.(*pp).fmtBytes(0xc11f3a3ee0, 0xc127fb1500, 0x63f, 0x700, 0x58, 0x2980362, 0xa)
	/usr/local/go/src/fmt/print.go:494 +0xc9d
fmt.(*pp).printValue(0xc11f3a3ee0, 0x2ac45e0, 0xc012f6b120, 0x97, 0x58, 0x0)
	/usr/local/go/src/fmt/print.go:845 +0x495
fmt.(*pp).printArg(0xc11f3a3ee0, 0x2ac45e0, 0xc012f6b120, 0x58)
	/usr/local/go/src/fmt/print.go:716 +0x2ef
fmt.(*pp).doPrintf(0xc11f3a3ee0, 0x2d5be59, 0x2, 0xc15e5a5720, 0x1, 0x1)
	/usr/local/go/src/fmt/print.go:1126 +0x913
fmt.Sprintf(0x2d5be59, 0x2, 0xc15e5a5720, 0x1, 0x1, 0x0, 0x6c2540)
	/usr/local/go/src/fmt/print.go:219 +0x74
github.com/cockroachdb/cockroach/pkg/geo/geopb.(*SpatialObject).EWKBHex(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/geo/geopb/geopb.go:17
github.com/cockroachdb/cockroach/pkg/geo.(*Geography).EWKBHex(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/geo/geo.go:495
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*DGeography).Format(0xc0a89d76b0, 0xc11f482240)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:2812 +0x103
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FmtCtx).formatNodeOrHideConstants(0xc11f482240, 0x3753b60, 0xc0a89d76b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/hide_constants.go:43 +0xcc
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FmtCtx).FormatNode(0xc11f482240, 0x3753b60, 0xc0a89d76b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:391 +0xbb
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.AsStringWithFlags(0x3753b60, 0xc0a89d76b0, 0x0, 0x2ceec00, 0x14923f5fb680)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:418 +0x6e
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.AsString(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:432
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*DGeography).String(0xc0a89d76b0, 0x68cc83, 0x2b352e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/expr.go:1781 +0x4e
github.com/cockroachdb/cockroach/pkg/col/coldata.AssertEquivalentBatches(0x379ecc0, 0xc03f730500, 0x37d0f80, 0xc069571560, 0x37d0f80, 0xc0f8016900)
	/go/src/github.com/cockroachdb/cockroach/pkg/col/coldata/testutils.go:105 +0xed8
github.com/cockroachdb/cockroach/pkg/sql/colcontainer_test.TestDiskQueue.func1(0xc03f730500)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colcontainer/diskqueue_test.go:135 +0xbd7
testing.tRunner(0xc03f730500, 0xc0ada23570)
	/usr/local/go/src/testing/testing.go:909 +0x19a
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x652
```

--- a/pkg/sql/colcontainer/diskqueue_test.go
+++ b/pkg/sql/colcontainer/diskqueue_test.go
@@ -59,7 +59,7 @@ func TestDiskQueue(t *testing.T) {
 					dequeuedProbabilityBeforeAllEnqueuesAreDone = 0
 					prefix, suffix = "Rewindable/", ""
 				}
-				numBatches := 1 + rng.Intn(1024)
+				numBatches := 1 + rng.Intn(16)
 				t.Run(fmt.Sprintf("%sDiskQueueCacheMode=%d/AlwaysCompress=%t%s/NumBatches=%d",
 					prefix, diskQueueCacheMode, alwaysCompress, suffix, numBatches), func(t *testing.T) {
 					// Create random input.

--- a/pkg/sql/colexec/spilling_queue_test.go
+++ b/pkg/sql/colexec/spilling_queue_test.go
@@ -55,7 +55,7 @@ func TestSpillingQueue(t *testing.T) {
 				dequeuedProbabilityBeforeAllEnqueuesAreDone = 0
 				prefix = "Rewindable/"
 			}
-			numBatches := 1 + rng.Intn(1024)
+			numBatches := 1 + rng.Intn(16)
 			log.Infof(context.Background(), "%sMemoryLimit=%s/DiskQueueCacheMode=%d/AlwaysCompress=%t/NumBatches=%d",
 				prefix, humanizeutil.IBytes(memoryLimit), diskQueueCacheMode, alwaysCompress, numBatches)
 			// Create random input.

